libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/scatterPlotUiConfigHelper.ts

```diff
@@ -35,5 +35,26 @@ export function setScatterPlotUiConfig(
     set(referencePointConfigured, [UICONFIG, BUCKETS, BucketNames.ATTRIBUTE, "icon"], scatterViewIcon);
     set(referencePointConfigured, [UICONFIG, BUCKETS, BucketNames.SEGMENT, "icon"], scatterSegmentIcon);
 
+    const segmentByAttributes = getBucketItems(referencePoint.buckets, BucketNames.SEGMENT);
+
+    if (
+        segmentByAttributes.length > 0 &&
+        referencePointConfigured.properties?.controls?.clustering?.enabled
+    ) {
+        referencePointConfigured = {
+            ...referencePointConfigured,
+            properties: {
+                ...referencePointConfigured.properties,
+                controls: {
+                    ...referencePointConfigured.properties.controls,
+                    clustering: {
+                        ...referencePointConfigured.properties.controls.clustering,
+                        enabled: false,
```

Comment:
Is this correct? Do you specifically want to overwrite enablement?

Reply:
Yes, when you drop segmentBy bucket, clustering should be always disabled.